deepmd/utils/data.py

```diff
@@ -61,15 +61,22 @@ def __init__(
         root = DPPath(sys_path)
         self.dirs = root.glob(set_prefix + ".*")
         self.dirs.sort()
-        self.mixed_type = self._check_mode(
-            self.dirs[0]
-        )  # mixed_type format only has one set
+        # check mix_type format
+        error_format_msg = (
+            "if one of the set is of mixed_type format, "
+            "then all of the sets in this system should be of mixed_type format!"
+        )
+        if self._check_mode(self.dirs[0]):
+            for set_item in self.dirs[1:]:
+                assert self._check_mode(set_item), error_format_msg
+            self.mixed_type = True
+        else:
+            for set_item in self.dirs[1:]:
+                assert not self._check_mode(set_item), error_format_msg
+            self.mixed_type = False
```

Comment:
Here two conditions can be merged
```suggestion
        self.mixed_type = self._check_mode(self.dirs[0]):
        for set_item in self.dirs[1:]:
            assert self._check_mode(set_item) == self.mixed_type, error_format_msg
```